File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        if len(prices) == 1:
            return 0
        
        ans = []
        sellpoint = 0
        buypoint = 10 ** 4
        for i in range(len(prices)):
            if buypoint > prices[i]:
                buypoint = prices[i]
                sellpoint = 0
            if sellpoint < prices[i]:
                sellpoint = prices[i]
                ans.append(sellpoint - buypoint)

        return max(ans)
            

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.c
int maxProfit(int* prices, int pricesSize) {
    if (pricesSize == 1) return 0;
        
    int ans = 0;
    int sellpoint = 0;
    int buypoint = 10000;

    for (int i=0;i<pricesSize;i++) {
        if (buypoint > prices[i]) {
            buypoint = prices[i];
            sellpoint = 0;
        }
        if (sellpoint < prices[i]) {
            sellpoint = prices[i];
            if (ans < sellpoint - buypoint) {
                ans = sellpoint - buypoint;
            }
        }
    }
    return ans;
}